### PR TITLE
Insertion sort in Rust

### LINKED ISSUE #2148: Insertion Sort
- [x] C
- [x] C++
- [x] C#
- [x] CoffeeScript
- [x] Dart
- [x] Go
- [x] Java
- [x] JS
- [x] Kotlin
- [ ] Php - @saloniankita 
- [x] Python
- [ ] Readme - @debdutgoswami
- [x] Ruby
- [x] Rust - @m-code12
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by m-code12**
Fixes #2148 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added Insertion Sort in Rust

#### Languages Used:
- Rust

#### Files Added:
- Insertion_Sort.rs

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by m-code12** [comment]
I have created this pull request following all the required coding conventions.Please review and merge @singh-shreya6

File: Insertion_Sort/Insertion_Sort.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+use std::io;
+use std::str::FromStr;
+
+fn main() {
+    println!("Enter the numbers to be sorted (separated by space) : ");
+    let mut i = read_values::<i32>().unwrap();
+    insertion_sort(&mut i);
+    println!("Sorted array: {:?}", i)
+    }
```
Comment (singh-shreya6) [inline Insertion_Sort/Insertion_Sort.rs:9]: This bracket does not need indentation. Please shift this to fall in line with f of fn main()

File: Insertion_Sort/Insertion_Sort.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
+use std::io;
+use std::str::FromStr;
+
+fn main() {
+    println!("Enter the numbers to be sorted (separated by space) : ");
+    let mut i = read_values::<i32>().unwrap();
```
Comment (singh-shreya6) [inline Insertion_Sort/Insertion_Sort.rs:6]: please add space around ::

File: Insertion_Sort/Insertion_Sort.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    insertion_sort(&mut i);
+    println!("Sorted array: {:?}", i)
+    }
+
+fn insertion_sort(arr: &mut Vec<i32>) {
+    for i in 1..arr.len()  {
+        let mut j = i;
+        while j > 0 && arr[j] < arr[j-1] {
+            arr.swap(j, j-1);
```
Comment (singh-shreya6) [inline Insertion_Sort/Insertion_Sort.rs:15]: Space around -

File: Insertion_Sort/Insertion_Sort.rs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    println!("Enter the numbers to be sorted (separated by space) : ");
+    let mut i = read_values::<i32>().unwrap();
+    insertion_sort(&mut i);
+    println!("Sorted array: {:?}", i)
+    }
+
+fn insertion_sort(arr: &mut Vec<i32>) {
+    for i in 1..arr.len()  {
+        let mut j = i;
+        while j > 0 && arr[j] < arr[j-1] {
```
Comment (singh-shreya6) [inline Insertion_Sort/Insertion_Sort.rs:14]: Space around -

File: Insertion_Sort/Insertion_Sort.rs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            arr.swap(j, j-1);
+            j=j-1;
```
Comment (singh-shreya6) [inline Insertion_Sort/Insertion_Sort.rs:16]: Space around - and =

**Comment by m-code12** [comment]
@singh-shreya6 I have made all the required changes . Please review and merge.

File: Insertion_Sort/Insertion_Sort.rs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    let mut s = String::new();
+    io::stdin()
+        .read_line(&mut s)
+        .expect("could not read from stdin");
+    s.trim()
+        .split_whitespace()
+        .map(|word| word.parse())
+        .collect()
+}
+
```
Comment (asha15) [inline Insertion_Sort/Insertion_Sort.rs:31]: remove this line

File: Insertion_Sort/Insertion_Sort.rs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+Sample Input :
+    Enter the numbers to be sorted (separated by space) :                                                                         
+    45 34 112 50 91                                                                                                               
+
+Sample Output :
+    Sorted array: [34, 45, 50, 91, 112]
+
+*/
+
```
Comment (asha15) [inline Insertion_Sort/Insertion_Sort.rs:43]: remove this line too

**Comment by m-code12** [comment]
@asha15 I have made the changes as you had said. Please review and merge.

File: Insertion_Sort/Insertion_Sort.rs [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+/*
+
+Sample Input :
+    Enter the numbers to be sorted (separated by space) :                                                                         
+    45 34 112 50 91                                                                                                               
+
+Sample Output :
+    Sorted array: [34, 45, 50, 91, 112]
+
+*/
```
Comment (singh-shreya6) [inline Insertion_Sort/Insertion_Sort.rs:41]: Please keep a blank line at end of file. Otherwise it shows as error.

**Comment by m-code12** [comment]
@singh-shreya6 made the changes.

**Pull request merged**